source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -2371,3 +2372,40 @@ message TransformationWrapRegionInSelection {
   bool branch_condition = 3;
 
 }
+
+message TransformationWrapVectorSynonym {
+  // A transformation that wraps an arithmetic operation into a vector operation
+  // and get the result of the original operation from the corresponding index.
+  // For instance, after the transformation, an addition operation between two scalars:
+  // c = a + b
+  //
+  // will be wrapped into vectors:
+  //
+  // va = vector(..., a, ...)
+  // vb = vector(..., b, ...)
+  // vc = va + vb
+  //
+  // where the result will be updated from the corresponding index of the result vector:
+  //
+  // c = vc[corresponding_idx]
+
+  // The result if of the original scalar operation instruction.
+  uint32 instruction_id = 1;
+
+  // The result id for the first vector that contains the first value of the scalar operation.
+  uint32 result_id1 = 2;
+
+  // The result id for the second vector that contains the second value of the scalar operation.
+  uint32 result_id2 = 3;
+
+  // A fresh id for the resulted vector from the addition of the first and second vector.
+  uint32 vec_id = 4;
+
+  // Id of the vector type. Must be a valid type that is either vec2, vec3 or vec4.
+  uint32 vec_type_id = 5;
```

Comment:
I suggest removing this field from the message, as it can be derived from the other fields: you can require that the types of the two given vectors match, and that they are compatible with the type of the operands to the scalar instruction.

Reply:
Done.